src/main/java/org/apache/mesos/scheduler/plan/DefaultInstallStrategy.java

```diff
@@ -18,12 +19,12 @@ public DefaultInstallStrategy(Phase phase) {
     }
 
     @Override
-    public Optional<Block> getCurrentBlock() {
+    public Optional<Block> getCurrentBlock(Collection<String> dirtiedAssets) {
         if (interrupted.get()) {
             return Optional.empty();
         } else {
             for (Block block : phase.getBlocks()) {
-                if (!block.isComplete()) {
+                if (!block.isComplete() && !dirtiedAssets.contains(block.getName())) {
```

Comment:
`getCurrentBlock()` is a public interface method. `dirtiedAssets` can be `null`. Maybe validate ? `... !CollectionUtils.isNotEmpty(dirtiedAssets) ...`